librz/core/cmd/cmd_print.c

```diff
@@ -2564,6 +2547,221 @@ static void core_print_raw_buffer(RzStrStringifyOpt *opt) {
+		RZ_LOG_ERROR("core: the block size is too small to read string (expected at least %u but got %u bytes).\n", core->blocksize, min_size);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	RzCmdStatus status = RZ_CMD_STATUS_ERROR;
+	if (*core->block & 0x1) { // "long" string
+		const ut8 *ptr = core->block + (bitness / 8) * 2;
+		ut64 old_offset = core->offset;
+		ut64 new_offset = rz_read_ble(ptr, big_endian, bitness);
+
+		rz_core_seek(core, new_offset, SEEK_SET);
+		rz_core_block_read(core);
+
+		status = core_auto_detect_and_print_string(core, true, 0, mode);
+
+		rz_core_seek(core, old_offset, SEEK_SET);
+		rz_core_block_read(core);
+	} else {
+		status = core_auto_detect_and_print_string(core, true, 1, mode);
+	}
+
+	return status;
+}
+
+RZ_IPI RzCmdStatus rz_print_strings_current_block_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzListIter *it = NULL;
+	RzDetectedString *detected = NULL;
+	RzUtilStrScanOptions scan_opt = {
+		.buf_size = core->blocksize,
+		.max_uni_blocks = 4,
+		.min_str_length = core->bin->minstrlen,
+		.prefer_big_endian = false,
+	};
+
+	RzList *found = rz_list_newf((RzListFree)rz_detected_string_free);
+	if (!found) {
+		RZ_LOG_ERROR("core: failed to allocate RzList\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (rz_scan_strings_raw(core->block, found, &scan_opt, 0, core->blocksize, RZ_STRING_ENC_GUESS) < 0) {
+		rz_list_free(found);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	rz_list_foreach (found, it, detected) {
+		ut64 address = core->offset + detected->addr;
+		if (mode != RZ_OUTPUT_MODE_QUIET) {
+			rz_print_offset(core->print, address, 0, 0, 0, 0, NULL);
+		}
+		rz_cons_memcat(detected->string, detected->size);
+		rz_cons_newline();
+	}
+
+	rz_list_free(found);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_first_string_current_block_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzDetectedString *detected = NULL;
+	RzUtilStrScanOptions scan_opt = {
+		.buf_size = core->blocksize,
+		.max_uni_blocks = 4,
+		.min_str_length = core->bin->minstrlen,
+		.prefer_big_endian = false,
+	};
+
+	RzList *found = rz_list_newf((RzListFree)rz_detected_string_free);
+	if (!found) {
+		RZ_LOG_ERROR("core: failed to allocate RzList\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	if (rz_scan_strings_raw(core->block, found, &scan_opt, 0, core->blocksize, RZ_STRING_ENC_GUESS) < 0) {
+		rz_list_free(found);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	detected = rz_list_first(found);
+	if (detected) {
+		rz_cons_memcat(detected->string, detected->size);
+		rz_cons_newline();
+	}
+
+	rz_list_free(found);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_pascal_string_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzStrStringifyOpt opt = { 0 };
+	bool big_endian = rz_config_get_b(core->config, "cfg.bigendian");
+	ut64 string_len = 0;
+	ut32 offset = 0;
+
+	if (!strcmp(argv[1], "8")) {
+		string_len = (ut64)core->block[0];
```

Comment:
Will it work for BE platforms too?

Reply:
yes, seems that the format is like that, either first byte, half-word, word, quad-word